packages/legend-query-builder/src/components/fetch-structure/QueryBuilderResultModifierPanel.tsx

```diff
@@ -380,6 +385,283 @@ export const QueryResultModifierModal = observer(
+                businessDate,
+                setBusinessDate,
+              )(ps) &&
+              shouldFilterMilestoningParamIfNotUsed(
+                processingDate,
+                setProcessingDate,
+              )(ps) &&
+              shouldFilterMilestoningParamIfNotUsed(
+                startDate,
+                setStartDate,
+              )(ps) &&
+              shouldFilterMilestoningParamIfNotUsed(endDate, setEndDate)(ps),
+          ),
+        ]);
+      } else {
+        //get or initialize getAll() parameters
+        setIsAllVersionsInRangeEnabled(false);
+        let newParamStates: LambdaParameterState[] = [];
+        if (
+          (!businessDate || !processingDate) &&
+          milestoningState.queryBuilderState.class
+        ) {
+          const stereotype = getMilestoneTemporalStereotype(
+            milestoningState.queryBuilderState.class,
+            milestoningState.queryBuilderState.graphManagerState.graph,
+          );
+          if (stereotype) {
+            const existingParamStates = parameterStates.map(
+              (ps) => ps.variableName,
+            );
+            newParamStates = milestoningState
+              .getMilestoningImplementation(stereotype)
+              .buildParameterStatesFromMilestoningParameters()
+              .filter((ps) => !existingParamStates.includes(ps.parameter.name));
+          }
+        }
+        const allParamStates = [...newParamStates, ...parameterStates].filter(
+          (ps) =>
+            shouldFilterMilestoningParamIfNotUsed(
+              startDate,
+              setStartDate,
+            )(ps) &&
+            shouldFilterMilestoningParamIfNotUsed(endDate, setEndDate)(ps),
+        );
+        setParameterStates(allParamStates);
+        if (!businessDate) {
+          setBusinessDate(
+            allParamStates.find(
+              (ps) =>
+                ps.variableName ===
+                (milestoningState.businessDate &&
+                milestoningState.businessDate instanceof VariableExpression
+                  ? milestoningState.businessDate.name
+                  : BUSINESS_DATE_MILESTONING_PROPERTY_NAME),
+            )?.parameter,
+          );
+        }
+        if (!processingDate) {
+          setProcessingDate(
+            allParamStates.find(
+              (ps) =>
+                ps.variableName ===
+                (milestoningState.businessDate &&
+                milestoningState.processingDate instanceof VariableExpression
+                  ? milestoningState.processingDate.name
+                  : PROCESSING_DATE_MILESTONING_PROPERTY_NAME),
+            )?.parameter,
+          );
+        }
+      }
+      setIsAllVersionsEnabled(Boolean(value));
+    };
+
+    const buildAllVersionsInRangeParameters = (): void => {
+      let startDateParameterState;
+      let endDateParameterState;
+      if (!(startDate && startDate instanceof VariableExpression)) {
+        const startDateVar = new VariableExpression(
```

Comment:
Does it mean we don't let users to have hardcoded startDate and endDate?

Reply:
https://github.com/finos/legend-studio/assets/73408381/a8d0b9eb-d6e7-442d-8994-656aca39ff20

This is the demo video for the case where the root class could be set to use allVersionsInRange 


https://github.com/finos/legend-studio/assets/73408381/c4f482e8-cee8-445f-9c86-959e46e3fa9f